### PR TITLE
Feat: Adds on demand taxi routes

### LINKED ISSUE #350: Booking On-Demand Transport (Taxi)
In Herrenberg after 9pm some busses are replaced with Taxis that you need to book in advance.

For example, when you want to travel from Nufringen to Gültstein at 10 pm, you will receive something that looks like a normal bus route: https://tinyurl.com/yds6qdkz

Screenshot from 2021-05-31 09-26-21

## Flex mode

However, you can also book a special mode that will take you directly to your door (if it's within 300m of a bus stop). We want to show this mode in a separate "tile" next to the cycling, p+r ... modes. 
This is the "Taxi" Icon that we are using combined with the public transit icon for the new flex mode tile: 
[TaxiIcon.zip](https://github.com/stadtnavi/digitransit-ui/files/6605014/TaxiIcon.zip)

Screenshot from 2021-05-31 10-04-01

The icon should be a car with a bus in the background.

In this tile, you need to add the modes `FLEX_DIRECT`, `FLEX_EGRESS`. The result should then look like this.

Screenshot from 2021-05-31 10-09-40

## Booking info

Also in both cases you must get additional data from the GraphQL API about the following booking requirements:

- booking telephone number and URL
- a warning about having to book ahead of time

We have added new GraphQL properties for the booking info and with the following query you can get the message, telephone number and URL.

URL: https://api.dev.stadtnavi.eu/otp/routers/default/index/graphql

```
query{
  plan(
    date: "2021-06-02"
    time: "22:30"
    from: { lat:  48.6226, lon: 8.8879 }
    to: { lat: 48.5755, lon: 8.8795 }
    transportModes: [
      { mode: RAIL },
      { mode: FLEX, qualifier: EGRESS },
      { mode: FLEX, qualifier: DIRECT },
      { mode: WALK },
    ],
    maxWalkDistance: 5000
  ) {
    itineraries {
      legs {
        mode
        rentedBike
        distance
        startTime
        endTime
        pickupBookingInfo {
          message
          contactInfo {
            phoneNumber
            infoUrl
            bookingUrl
          }
        }
        route {
          url
          mode
          shortName
        }
        trip {
          gtfsId
          tripShortName
        }
        from {
          name
          bikeRentalStation {
            name
            id
          }
        }
        to {
          name
          bikeRentalStation {
            name
            id
          }
        }
      }
    }
  }
}
```

The new part is:

```
 pickupBookingInfo {
          message
          contactInfo {
            phoneNumber
            infoUrl
            bookingUrl
          }
        }
```

### PR DESCRIPTION
**Pull request by andreashelms**
## Proposed Changes

image
image

If there is a bookinUrl set in graphql response UI will look like that:

image

## Review

  - Read and verify the code changes
  - Test the functionality by running the UI locally with all popular browsers available in your platform

closes #350

### CODE REVIEW

**Comment by leonardehrenfried** [comment]
@Denis306 has also provided the taxi icon here: https://github.com/stadtnavi/digitransit-ui/files/6605014/TaxiIcon.zip

(It was in the original ticket but I had overlooked it, too)

Can you please use that?

File: app/util/planParamUtil.js [hunk 1]
```diff
@@ -190,6 +190,13 @@ const getShouldMakeCarQuery = (
   );
 };
 
+const getShouldMakeOnDemandTaxiQuery = time => {
+  const date = new Date(time * 1000);
+  return (
+    date.getHours() > 22 || (date.getHours() === 22 && date.getMinutes() > 0)
```
Comment (andreashelms) [inline app/util/planParamUtil.js]: @leonardehrenfried What is the desired end-time for executing on demand taxi queries?

Comment (leonardehrenfried) [reply app/util/planParamUtil.js]: 5 AM.

And also please start it at 9PM.

Comment (andreashelms) [reply app/util/planParamUtil.js]: Ok. But if I execute queries with departure time before 22:01 api doesn't return any results.

Comment (leonardehrenfried) [reply app/util/planParamUtil.js]: Hmm, maybe I'm wrong then. I will ask the experts. @derhuerst @hbruch

Comment (derhuerst) [reply app/util/planParamUtil.js]: There isn't a single time where the on-demand traffic starts, it varies by line and day.

For example, on Sundays, the [779 line](https://stadtwerke.herrenberg.de/fileadmin/user_upload/www/OEPNVUndParken/OEPNV/Fahrplaene/Fahrplaene_2021/Linie_779.pdf) is on-demand from 19:51, but on weekdays from 21:05; The 780 line on the other hand has fixed (*not* on-demand) runs until 10:30.

Flex routing still uses fixed (*not* on-demand) runs, and this is only a matter of performance, right? To be on the safe side, we could use 9pm and 5am, as proposed by Leonard.

**Comment by andreashelms** [comment]
@leonardehrenfried Done
image

**Comment by leonardehrenfried** [comment]
Thanks - well done!

**Pull request merged**